src/main/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessor.java

```diff
@@ -367,23 +367,26 @@ private void updateCloudWatchAttemptLogInformation(String fileName,
             logger.atTrace().kv("log-line-size-bytes", dataSize)
                     .log("Log line larger than maximum event size 256KB, " + "will be split into multiple log events");
         }
-        int totalChunks = dataSize / MAX_EVENT_LENGTH + 1;
+        int totalChunks = (int) Math.ceil((float) dataSize / MAX_EVENT_LENGTH);
```

Comment:
I think you could use `(dataSize - 1)/ MAX_EVENT_LENGTH + 1` to avoid casting and float division, assuming dataSize nonzero. Although it's a little intuitive so you might also need to add comments.